### PR TITLE
implement api book detail

### PR DESCRIPTION
**Pull request by ghost**
## Description
return book info: Image, name, Author, year, page number, price, description, review score
## Reference Requirements
- Product Backlog ID: PBI-38
- Task ID: #38.1, #38.2

## Reference Isuees
#{issue_id}

## Migrations
YES

## Impacted Areas in Application
List general components of the application that this PR will affect:

- No

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
- 
SELECT name, author, year, pages, price, description, image, avg_rating
FROM books;
- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)
image

### CODE REVIEW

File: app/Http/Controllers/Api/BookController.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            'price',
+            'image',
+            'description',
+            'avg_rating'
+        ];
+
+        $books = $this->book->select($fields)->findOrFail($id);
+        return response()->json([
+            'books' => $books,
+            'success' => true
```
Comment (at-annguyen) [inline app/Http/Controllers/Api/BookController.php]: code number return? Put it in meta.

**Review by annq3012 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Api/BookController.php [hunk 2]
```diff
@@ -18,6 +18,53 @@
 
 class BookController extends Controller
 {
+    /**
+     * The Book implementation.
+     *
+     * @var Book
+     */
+    protected $book;
+    /**
```
Comment (at-annguyen) [inline app/Http/Controllers/Api/BookController.php]: 1 line

File: app/Http/Controllers/Api/BookController.php [hunk 3]
```diff
@@ -18,6 +18,53 @@
...
+     * Create a new controller instance.
+     *
+     * @param Book $book instance of Book
+     *
+     * @return void
+     */
```
Comment (at-annguyen) [inline app/Http/Controllers/Api/BookController.php]: 1 line

File: app/Http/Controllers/Api/BookController.php [hunk 4]
```diff
@@ -18,6 +18,53 @@
...
+    public function __construct(Book $book)
+    {
+        $this->book = $book;
+    }
```
Comment (at-annguyen) [inline app/Http/Controllers/Api/BookController.php]: 1 line

File: app/Http/Controllers/Api/BookController.php [hunk 5]
```diff
@@ -18,6 +18,53 @@
...
+            'image',
+            'description',
+            'avg_rating'
+        ];
+        $books = $this->book->select($fields)->findOrFail($id);
+        
+        return response()->json([
+            "meta" => [
+                "status" => "successfully",
+                "code" => 200
```
Comment (at-annguyen) [inline app/Http/Controllers/Api/BookController.php]: 200 => Response::HTTP_OK

**Review by annq3012 — APPROVED** [review]

**Comment by at-annguyen** [comment]
LGTM

**Pull request merged**